pkg/controller/common/constants.go

```diff
@@ -4,6 +4,9 @@ const (
 	// GeneratedByControllerVersionAnnotationKey is used to tag the machineconfigs generated by the controller with the version of the controller.
 	GeneratedByControllerVersionAnnotationKey = "machineconfiguration.openshift.io/generated-by-controller-version"
 
+	// ReleaseImageVersionAnnotationKey is used to tag the rendered machineconfigs & controller config with the release image version.
+	ReleaseImageVersionAnnotationKey = "machineconfiguration.openshift.io/release-image-version"
```

Comment:
One big concern I had is checking whether this would cause the opposite problem of us re-rolling all nodes for an upgrade, even if the actual machineconfig content didn't change.

Because that the MCO rolls out changes to the pool based on the unique rendered config digest, so if we included the release image version in the digest, it'd cause that problem.

But, the digest (md5) is only of `config.Spec`, not the rest.  (And it really has to work that way because of the other similar annotation keys, etc.)

You and/or others may know this, I'm just writing this down for my own sanity really :smile:

Reply:
I went thru the same thing working on this if it helps :laughing: